src/pages/index.js

```diff
@@ -1,107 +1,244 @@
+        deleteConfirmationPopup.open(cardData._id);
+      });
+
+    return cardElement; // Return the correctly initialized card element
   }
 
-  // Handle image clicks to open the image preview popup
   function handleImageClick(data) {
-    popupWithImage.open(data); // Open image popup with data (name and link)
+    popupWithImage.open(data);
   }
 
-  // Handle opening the profile edit modal and populating it with current user data
   profileEditButton.addEventListener("click", () => {
     const userData = userInfo.getUserInfo();
     nameInput.value = userData.name;
     descriptionInput.value = userData.job;
-    profileEditPopup.open(); // Open the modal
-    profileFormValidator.resetValidation(); // Reset form validation for profile edit popup
+    profileEditPopup.open();
+    profileFormValidator.resetValidation();
   });
 
-  // Handle opening the modal for adding a new card
   addButton.addEventListener("click", () => {
-    addCardFormValidator.disableButton(); // Only disable button when opening add card popup
-    addCardPopup.open(); // Open the modal
+    addCardFormValidator.disableButton();
+    addCardPopup.open();
   });
 
-  /* Render all initial cards from initialCards array */
-  cardSection.renderItems();
+  // Update profile info
+  api
+    .getUserInfo()
+    .then((userData) => {
+      userInfo.setUserInfo({
+        name: userData.name,
+        job: userData.about,
+      });
+      const avatarImage = document.querySelector(".profile__image");
+      avatarImage.src = userData.avatar;
+      avatarImage.alt = userData.name;
+    })
+    .catch((error) => {
+      console.error("Error fetching user information:", error);
+    });
+
+  // Fetch initial cards
+  api
+    .getInitialCards()
+    .then((cards) => {
+      cards.forEach((cardData) => {
+        const cardElement = createCard(cardData, api); // Pass api to Card
+        cardSection.addItem(cardElement);
+      });
+    })
+    .catch((error) => {
+      console.error("Error fetching cards:", error);
+    });
+
+  // Handling avatar modal and avatar update
+  // Initialize the avatar modal with the Popup class
+  const avatarPopup = new PopupWithForm("#profile-avatar-modal", (formData) => {
+    const newAvatarUrl = formData.avatar;
+    console.log("Avatar URL is submitted:", newAvatarUrl);
+
+    // Change the button text to "Saving..."
+    const saveButton = document.querySelector("#modal-avatar-button");
+    saveButton.textContent = "Saving...";
+
+    // Send PATCH request to update avatar
+    console.log("Data sent to API:", { avatar: newAvatarUrl });
+    api
+      .updateProfileAvatar({ avatar: newAvatarUrl })
+      .then((userData) => {
+        const avatarImage = document.querySelector(".profile__image");
+        avatarImage.src = userData.avatar; // Update the avatar image
+        avatarImage.alt = "Profile Avatar"; // Update alt text
+        avatarPopup.close(); // Use the Popup class to close the modal
```

Comment:
* Please, **disable** the `avatar` save button after submission https://snipboard.io/v5Ymxp.jpg